crates/lib/kajiya-backend/src/vulkan/shader.rs

```diff
@@ -19,7 +19,10 @@ use std::{
 };
 
 pub const MAX_DESCRIPTOR_SETS: usize = 4;
+#[cfg(feature = "ray-tracing")]
 pub const MAX_BINDLESS_DESCRIPTOR_COUNT: usize = 512 * 1024;
+#[cfg(not(feature = "ray-tracing"))]
+pub const MAX_BINDLESS_DESCRIPTOR_COUNT: usize = 96;
```

Comment:
This should really be based off of `max_per_stage_descriptor_sampled_images` (128 on my macbook). Ideally something like `max_per_stage_descriptor_sampled_images - 32` or so.

Reply:
Yup, agreed that this should be dynamic, and have some runtime checks. It's also not tied to ray-tracing either -- all texturing in kajiya is done via bindless resources, so even the basic rasterizer will be affected by those limits.

How do you feel about plumbing through those limits dynamically? Should we do it here, or in a separate change?